app/models/filter.rb

```diff
@@ -0,0 +1,29 @@
+module Filter
+  def self.sort_filter(params)
+    developers = sort_developers(params)
+    tags = selected_tags(params)
+    preload = developers.joins(:tags)
+
+    tags.reduce(developers) do |acc, tag|
+      acc & tag.developers
+    end
+  end
+
+  def self.sort_developers(params)
+    sort[params["Sort By"]][Developer]
+  end
+
+  def self.sort
+    {
+      "Rate: Ascending" => lambda { |active| active.order(rate: :asc) },
+      "Rate: Descending" => lambda { |active| active.order(rate: :desc) }
+    }
+  end
+
+  def self.selected_tags(params)
+    params["filter"] ||= {}
+    params["filter"].select{ |key, val| val == "1" }.keys.map do |tag_name|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.<br>Space missing to the left of {.